File: Longest Increasing Subsequence/longestIncreasingSubsequence.cpp
#include<bits/stdc++.h>
using namespace std;
int longestIncreasingSubsequence(int n, int arr[]) {
	// Function to find the Longest Increasing Subsequence of the given sequence
	int dp[n + 1];

	// Every Element is LIS on it's own so setting value of dp to be 1
	for (int i = 0; i <= n; i++)
		dp[i] = 1;

	for (int i = 0; i < n; i++) {
		for (int j = i; j < n; j++) {
			if (arr[j] > arr[i])
				dp[j] = max(dp[i] + 1, dp[j]);
		}
	}
	int result = 0;
	for (int  i = 0; i < n; i++) {
		result = max(result, dp[i]);
	}
	return result;
}
int main() {
	int n = 9;
	int arr[] = {10, 22, 9, 33, 21, 50, 41, 60, 80};
	cout << longestIncreasingSubsequence(n, arr) << endl;
	return 0;
}

File: Longest Increasing Subsequence/longestIncreasingSubsequence.java
import java.util.*;
import java.lang.*;
import java.io.*;


class Main{
    //This function finds out the longest Increasing subsequence in an array.
    public static int longestIncreasingSubsequence(int[] arr,int n){
        int[] dp = new int[n + 1];
        Arrays.fill(dp,1);
        for (int i=0;i<n;i++){
            for (int j=i;j<n;j++){
                if (arr[j] > arr[i]) dp[j] = Math.max(dp[i] + 1, dp[j]);
            }
        }
        int max = 0;
        for (int i=0;i<n;i++){
            max = Math.max(dp[i],max);
        }
        return max;
    }
    //Main Function
    public static void main(String[] args){
        int n = 9;
        int[] arr = {10, 22, 9, 33, 21, 50, 41, 60, 80};
        int ans = longestIncreasingSubsequence(arr,n);
        System.out.println(ans);
    }
}
